mono/metadata/class.c

```diff
@@ -6312,66 +6375,70 @@ set_anon_gparam_class (MonoGenericParam *param, gboolean is_mvar, MonoClass *kla
  * LOCKING: Acquires the image lock (@image).
  */
 MonoClass *
-mono_class_from_generic_parameter (MonoGenericParam *param, MonoImage *image, gboolean is_mvar)
+mono_class_from_generic_parameter_internal (MonoGenericParam *param)
 {
-	MonoGenericContainer *container = mono_generic_param_owner (param);
-	MonoGenericParamInfo *pinfo = NULL;
-	MonoClass *klass, *klass2;
+	MonoImage *image = get_image_for_generic_param (param);
+	MonoGenericParamInfo *pinfo = mono_generic_param_info (param);
+	MonoClass *klass, *klass_other_thread;
```

Comment:
In general, we don't rename variables unless they have the wrong name, it destroys history more than helping the code.

In this case is ok as all uses had to be changed anyways.